docs/decisions/0033-store-chats-in-mvstore.md

```diff
@@ -0,0 +1,49 @@
+---
+nav_order: 0033
+parent: Decision Records
+---
+# Store chats in MVStore
+
+## Context and Problem Statement
+
+This is a follow-up to [ADR-031](0032-store-chats-in-local-user-folder).
+
+The chats with AI should be saved on exit from JabRef and retrieved on launch. We need to decide the format of
+the serialized messages.
+
+## Decision Drivers
+
+* Easy to implement and maintain
+* Memory-efficient (because JabRef is said to consume much memory)
+
+## Considered Options
+
+* JSON
+* MVStore
+* Custom format
+
+## Decision Outcome
+
+Chosen option: "MVStore", because it is simple and memory-efficient.
+
+## Pros and Cons of the Options
+
+### JSON
+
+* Good, because allows for easy storing and loading of chats
+* Good, because cross-platform
+* Good, because widely used and accepted, so there are lots of libraries for JSON format
+* Good, because it is even possible to reuse the chats file for other purposes
+* Bad, because too verbose (meaning the file size could be much smaller)
+
+### MVStore
+
+* Good, because automatic loading and saving to disk
+* Good, because memory-efficient
+* Bad, because the order of messages need to be "hand-crafted" (e.g., by mapping from an Integer to the concrete message)
```

Comment:
Unfortunately, I switched to `langchain4j` chat message type, which is not serializable.

So, I switched to old version. This line should left

Reply:
Even then, you can use `List`. Then, the contents of the list are your custom elements to be mapped to `chat message type`. You need to define a `record` holding the elements.

I think, its about src/main/java/org/jabref/logic/ai/chathistory/BibDatabaseChatHistoryManager.java. After quickly browsing through, a `record` should do the job.

(Please provide links next time. Here on my laptop screen, the scroll bar is large and I cannot find the source you are referring to, easily. I need links to be able to follow quickly. Otherwise, my time runs out. Unfortunately, I do not have unlimited time for reviewing and fixing the Java eco system. If I have links to `chat message type` and to the class, it makes my life (and the one of other reviewers) more easy. It frees time for us to comment on more things. I know, this takes time on your side. I found the plugin https://github.com/kawamataryo/copy-git-link useful. Then, one can provide deep links to the class as Markdown from IntelliJ.)